eclair-core/src/test/scala/fr/acinq/eclair/channel/FuzzySpec.scala

```diff
@@ -81,107 +81,101 @@ class FuzzySpec extends TestkitBaseClass with StateTestsHelperMethods with Loggi
+            context become {
+              case u: UpdateFulfillHtlc =>
+                log.info(s"successfully sent htlc #${u.id}")
+                initiatePaymentOrStop(remaining - 1)
+              case u: UpdateFailHtlc =>
+                log.warning(s"htlc failed: ${u.id}")
+                initiatePaymentOrStop(remaining - 1)
+              case Status.Failure(t) =>
+                log.error(s"htlc error: ${t.getMessage}")
+                initiatePaymentOrStop(remaining - 1)
+            }
+        }
       } else {
-        paymentHandler ! ReceivePayment(Some(MilliSatoshi(requiredAmount)), "One coffee")
-        context become waitingForPaymentRequest
+        context stop self
+        latch.countDown()
       }
 
-    initiatePayment(false)
+    initiatePaymentOrStop(count)
 
     override def receive: Receive = ???
 
-    def waitingForPaymentRequest: Receive = {
-      case req: PaymentRequest =>
-        channel ! buildCmdAdd(req.paymentHash, req.nodeId)
-        context become waitingForFulfill(false)
-    }
-
-    def waitingForFulfill(stopping: Boolean): Receive = {
-      case u: UpdateFulfillHtlc =>
-        log.info(s"successfully sent htlc #${u.id}")
-        latch.countDown()
-        initiatePayment(stopping)
-      case u: UpdateFailHtlc =>
-        log.warning(s"htlc failed: ${u.id}")
-        initiatePayment(stopping)
-      case Status.Failure(t) =>
-        log.error(s"htlc error: ${t.getMessage}")
-        initiatePayment(stopping)
-      case 'stop =>
-        log.warning(s"stopping...")
-        context become waitingForFulfill(true)
-    }
-
   }
 
   test("fuzzy test with only one party sending HTLCs", Tag("fuzzy")) { f =>
     import f._
-    val latch = new CountDownLatch(100)
-    system.actorOf(Props(new SenderActor(alice, paymentHandlerB, latch)))
-    system.actorOf(Props(new SenderActor(alice, paymentHandlerB, latch)))
+    val senders = 2
+    val totalMessages = 100
+    val latch = new CountDownLatch(senders)
+    for (_ <- 0 until senders) system.actorOf(Props(new SenderActor(alice, paymentHandlerB, latch, totalMessages / senders)))
     awaitCond(latch.getCount == 0, max = 2 minutes)
-    assert(alice.stateName == NORMAL || alice.stateName == OFFLINE)
-    assert(bob.stateName == NORMAL || alice.stateName == OFFLINE)
+    assert(List(NORMAL, OFFLINE, SYNCING).contains(alice.stateName))
+    assert(List(NORMAL, OFFLINE, SYNCING).contains(bob.stateName))
   }
 
   test("fuzzy test with both parties sending HTLCs", Tag("fuzzy")) { f =>
     import f._
-    val latch = new CountDownLatch(100)
-    system.actorOf(Props(new SenderActor(alice, paymentHandlerB, latch)))
-    system.actorOf(Props(new SenderActor(alice, paymentHandlerB, latch)))
-    system.actorOf(Props(new SenderActor(bob, paymentHandlerA, latch)))
-    system.actorOf(Props(new SenderActor(bob, paymentHandlerA, latch)))
+    val senders = 2
+    val totalMessages = 100
+    val latch = new CountDownLatch(senders)
```

Comment:
Shouldn't the latch be `2*senders`?
Since you create `senders` actors with `alice` and `senders` actors with `bob`.

Reply:
Maybe `totalMessages` should also be halved (not sure)

Reply:
Or like you did in the other test, separate in two latches.